### ISSUE TITLE
False positive E0107 "wrong number of arguments" on variadic extern "C" function pointer calls

### ISSUE THREAD

**Issue by ldgabbay** [rust-lang/rust-analyzer#22573]
## Summary

rust-analyzer reports E0107 ("wrong number of arguments") when calling a
variadic `extern "C"` function *pointer* with only the named arguments.
`rustc` and `cargo check` accept the code without error.

## Reproduction

```rust
use std::os::raw::c_int;

#[repr(C)]
enum NifOption { DelayHalt }

struct NifEnv;

struct Funcs {
    pub set_option: unsafe extern "C" fn(*mut NifEnv, NifOption, ...) -> c_int,
}

fn funcs() -> &'static Funcs { todo!() }

fn main() {
    let env: *mut NifEnv = std::ptr::null_mut();
    unsafe { (funcs().set_option)(env, NifOption::DelayHalt) };
    // rust-analyzer: E0107 expected 3 arguments, found 2
    // rustc: no error
}
```

## Expected behaviour

No diagnostic. The `...` in a C-variadic function pointer is not a required
argument; calling with only the named parameters is valid.

## Actual behaviour

E0107: `expected 3 arguments, found 2` is reported on the call site.
rust-analyzer is counting the `...` slot as a required positional argument.

## Notes

This is distinct from #3088 (variadic function *declarations* parsed
incorrectly), which was fixed. This appears to be in the arity-checking
path for variadic function *pointers* specifically.

## Version
rust analyzer vscode extension 0.3.2929
rust 1.95.0
